vshard/storage/init.lua

```diff
@@ -265,6 +265,21 @@ local function local_call(func_name, args)
     return pcall(netbox_self_call, netbox_self, func_name, args)
 end
 
+local function master_call(replicaset, func, args, opts)
```

Comment:
The main problem is that `replicaset` call-methods were never documented to be used only for vshard functions. A user can call their own functions which might not return any errors or return errors colliding with vshard's by codes, or by types, or anything else.

This is possible via the public API using `router.route(bucket_id):call(...)`. Essentially, `replicaset` is like `netbox`. It doesn't care whether it is used with vshard or not. The places where this is violated are bad ugly cases, and are not so critical (just grep by `vshard.` in `replicaset.lua`).

The function `replicaset_locate_master` is kind of in the grey zone. It does call a vshard function so it can rely on which errors are returned and when. We could make replicaset rely on just being used with vshard, but for a start we would need to make all their `call` functions go through `vshard.storage.something` first.

Reply:
Thank you for explanation!